src/import/pull.c

```diff
@@ -281,15 +285,35 @@ static int parse_argv(int argc, char *argv[]) {
                         if (r < 0)
                                 return log_error_errno(r, "Failed to parse --settings= parameter '%s': %m", optarg);
 
-                        arg_settings = r;
+                        SET_FLAG(arg_pull_flags, PULL_SETTINGS, r);
                         break;
 
                 case ARG_ROOTHASH:
                         r = parse_boolean(optarg);
                         if (r < 0)
                                 return log_error_errno(r, "Failed to parse --roothash= parameter '%s': %m", optarg);
 
-                        arg_roothash = r;
+                        SET_FLAG(arg_pull_flags, PULL_ROOTHASH, r);
+
+                        /* If we were asked to turn off the root hash, implicitly also turn off the root hash signature */
```

Comment:
shouldn't turning off the verity file also turn off the roothash and signature? There's no use for a verity file without the corresponding hash, right?

Reply:
the recommended way for verity stuff is still to ship it in the image itself, i.e. as separate GPT partition in the same GPT partition table. Doing external verity files is mostly there to support @bluca, not because it was pretty ;-)

but regardless if you use external verity files or verity included in the GPT image, the roothash needs to be specified either way somehow, there's no mechanism to encode thta in the GPT image...

Reply:
Who's this bluca fella, I'll have a word about this ugliness :-D

I understand the use case for the external root hash + GPT, but the question was about the other way around: if the roothash is disabled, does it make sense to look for the verity file?

Reply:
hmm, good question... maybe. need to think about it

Reply:
No problem, and not a blocker from my point of view - just thinking out loud

Reply:
hmm, so i think turning off the `.roothash.p7s` stuff if you disable `.roothash` is pretty obvious: if you don't have a file, you can't want to validate it either.

But for the `.verity` stuff it's not that easy: we do support acquiring the root has by different means, for example by encoding it on the command line of nspawn or RootHash= in unit files, or by adding it to the files as xattrs. So while I figure in most cases people who disable the roothash stuff probably also don't want the verity stuff i think this is just a "most" and not "all". and if we are in doubt we probably should download the stuff if it exists unless the user explicitly turns that off.

or in other words: automatisms that aren't always right suck. hence I think we should leave this on, so that users explicitly disable the verity stuff if they don't want it.

Reply:
Yep, that makes sense to me. Thanks!